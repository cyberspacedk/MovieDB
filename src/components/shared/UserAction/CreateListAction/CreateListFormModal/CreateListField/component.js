import React from 'react';
import PropTypes from 'prop-types';
import { Input, Form } from 'antd';

const { Item } = Form;

const CreateListField = ({ field, form: { touched, errors }, ...props }) => (
  <Item
    validateStatus={touched[field.name] && errors[field.name] && 'error'}
    help={errors[field.name]}
  >
    <Input type="text" {...field} {...props} />
  </Item>
);

CreateListField.propTypes = {
  field: PropTypes.shape(PropTypes.object),
  form: PropTypes.shape({
    touched: PropTypes.object,
    errors: PropTypes.object,
  }),
};

export default CreateListField;
