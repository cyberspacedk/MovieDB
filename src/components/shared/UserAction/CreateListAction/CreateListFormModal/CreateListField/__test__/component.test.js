import React from 'react';
import { shallow } from 'enzyme';
import CreateListField from '../component';

describe('Component: CreateListField', () => {
  const props = {
    field: {
      name: 'name',
    },
    form: {
      touched: {
        name: '',
      },
      errors: {
        name: '',
      },
    },
  };

  const wrapper = shallow(<CreateListField {...props} />);

  it('Match its snapshot.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Match its snapshot. Validation errors', () => {
    const nextProps = {
      ...props,
      form: {
        touched: {
          name: 'required',
        },
        errors: {
          name: 'required',
        },
      },
    };
    const wrapper = shallow(<CreateListField {...nextProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
