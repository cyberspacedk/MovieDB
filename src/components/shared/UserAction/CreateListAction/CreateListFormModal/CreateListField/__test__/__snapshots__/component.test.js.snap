// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CreateListField Match its snapshot. 1`] = `
<FormItem
  hasFeedback={false}
  help=""
  validateStatus=""
>
  <Input
    disabled={false}
    name="name"
    type="text"
  />
</FormItem>
`;

exports[`Component: CreateListField Match its snapshot. Validation errors 1`] = `
<FormItem
  hasFeedback={false}
  help="required"
  validateStatus="error"
>
  <Input
    disabled={false}
    name="name"
    type="text"
  />
</FormItem>
`;
